tests/foreman/ui/test_discoveredhosts.py

```diff
@@ -1,31 +1,96 @@
 # -*- encoding: utf-8 -*-
 """Test class for Foreman Discovery"""
+from fauxfactory import gen_string, gen_mac
+from nailgun import client, entities
+from robottelo.config import conf
 from robottelo.decorators import stubbed
+from robottelo.helpers import get_server_credentials
+from robottelo import ssh
 from robottelo.test import UITestCase
+from robottelo.ui.session import Session
+from time import sleep
 
 
 class Discovery(UITestCase):
     """Implements Foreman discovery tests in UI."""
 
-    @stubbed()
-    def test_host_discovery_1(self):
-        """@Test: Discover a host in legacy mode by setting
-        "proxy.type=foreman" in PXE default
-
-        @Feature: Foreman Discovery
-
-        @Setup: Provisioning should be configured
-
-        @Steps: PXE boot a host/VM
-
-        @Assert: Host should be successfully discovered
-
-        @Status: Manual
+    name = gen_string("alpha")
+    image_path = '/var/lib/libvirt/images/{0}.img'.format(name)
+
+    def _pxe_boot_host(self, mac):
+        """PXE boot a unknown host"""
+        libvirt_server = 'qemu+tcp://{0}:16509/system'.format(
+            conf.properties['main.server.hostname'])
+        ssh.command('virt-install --hvm --network=bridge:virbr1, --mac={0} '
+                    '--pxe --name {1} --ram=1024 --vcpus=1 --os-type=linux '
+                    '--os-variant=rhel7 --disk path={2},size=10 --connect {3} '
+                    '--noautoconsole'
+                    .format(mac, self.name, self.image_path, libvirt_server))
+        sleep(30)
+
+    @classmethod
+    def setUpClass(cls):
+        """Steps to Configure foreman discovery
+
+        1. Build PXE default template
+        2. Create Organization/Location
+        3. Update Global parameters to set default org and location for
+        discovered hosts.
+        4. Enable auto_provision flag to perform discovery via discovery rules.
 
         """
-
-    @stubbed()
-    def test_host_discovery_2(self):
+        # Build PXE default template to get default PXE file
+        response = client.get(
```

Comment:
Can't we implement that things using Nailgun? (It is so nice if we can remove any reference to `client.xxxxxx`)